flink-end-to-end-tests/test-scripts/common_s3_operations.sh

```diff
@@ -58,7 +64,11 @@ function aws_cli_stop() {
 if [[ $AWSCLI_CONTAINER_ID ]]; then
   aws_cli_stop
 fi
-aws_cli_start
+aws_cli_start || aws_cli_start
```

Comment:
Will simple retry like this let the script be executed multiple times without proper cleanup of previous containers, given the aws_cli_start is not Idempotent, afaiu.

Reply:
I'm assuming that `docker run -d ...` with detach did not create a container if the exit code is non-zero. So failed `aws_cli_start` shouldn't leave something to be cleaned. But I can't find docs stating that.

Happy to remove the retry, getting the test to fail faster with a more obvious cause is an improvement.

Reply:
I've added in a failsafe to kill/remove if `CONTAINER_ID` is non-empty after the docker run fails.